<a href="https://colab.research.google.com/github/Diante830/Capstone-Project-2---Identifying-Key-Drivers-of-Revenue-for-Zara-Stores/blob/main/Capstone_Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Identifying Key Drivers of Revenue for Zara Stores

### Data Analysis & Business Insights Visualization Project

This project aims to analyze in-store sales data from Zara to uncover the key factors that drive product performance. By examining product categories, pricing, promotions, seasonal trends, and store placement, the project seeks to provide actionable insights that inform inventory planning, merchandising strategies, and marketing decisions.

Ultimately, the goal is to identify which products and placements generate the most sales, understand how pricing and promotions affect customer purchasing, and reveal patterns that help optimize retail performance.

---

#Business Insights & Questions To Solve

## Analysing Zara’s sales data will help to uncover key questions for better store performance:


### 1️ Which product placements lead to higher sales?
- **Focus:** `Product_Position` (Aisle, End-cap, Front of Store) vs `Sales_Volume`  
- **Value:** Helps with merchandising strategy for in-store display.



### 2️ Which products and categories drive the most sales?
- **Focus:** Identify top-selling products and categories (`Product_Category`, `Sales_Volume`, `Name`)  
- **Value:** Shows Zara which products to prioritize in inventory and marketing.



### 3️ How does pricing impact sales?
- **Focus:** Compare sales volume vs `Price`  
- **Value:** Reveals the optimal pricing range and whether high/low-priced items perform better.



### 4️ Do promotions boost sales?
- **Focus:** Compare sales for `Promotion = Yes` vs `No`  
- **Value:** Evaluates effectiveness of promotions and informs marketing strategy.



### 5️ How does seasonality affect product performance?
- **Focus:** `Seasonal` vs `Sales_Volume`  
- **Value:** Identifies seasonal trends, helps with inventory planning and seasonal campaigns.

## Dataset Overview

The dataset used in this analysis contains in-store sales information for products sold by `Zara`. It includes details about product categories, pricing, promotions, seasonal status, placement in stores, and sales volumes, allowing for a comprehensive analysis of factors influencing retail performance.

### Dataset Characteristics

- **Data size:** 252 rows × 16 columns  
- **Time period:** Snapshot of product data in 2024
- **Data types:** Combination of numerical (e.g., `Sales_Volume`, `Price`) and categorical (e.g., `Product_Category`, `Promotion`, `Product_Position`) variables  
- **Columns / Variables:** 16  

### Key Variables Used in the Analysis

- **Product_Category** – Category of the product (e.g., jackets, shirts)  
- **Sales_Volume** – Number of units sold, used to measure product performance  
- **Price** – Product price, used to evaluate pricing impact on sales  
- **Promotion** – Indicates if the product was on promotion, used to assess marketing effectiveness  
- **Seasonal** – Shows whether the product is seasonal or year-round, used to study seasonal trends  
- **Product_Position** – Placement in the store (Aisle, End-cap, Front of Store), used to examine merchandising impact  
- **Name** – Product name, used to identify top-selling items

In [2]:
import pandas as pd

df = pd.read_csv('Zara_Sales_Analysis - Cleaned.csv')
df

,Product_ID,Product_Position,Promotion,Product_Category,Seasonal,Sales_Volume,brand,url,sku,name,description,price,currency,scraped_at,terms,section
0,185102,Aisle,No,Clothing,No,2823,Zara,https://www.zara.com/us/en/basic-puffer-jacket...,272145190-250-2,BASIC PUFFER JACKET,Puffer jacket made of tear-resistant ripstop f...,£19.99,USD,2024-02-19 08:50:06,jackets,MAN
1,188771,Aisle,No,Clothing,No,654,Zara,https://www.zara.com/us/en/tuxedo-jacket-p0889...,324052738-800-46,TUXEDO JACKET,Straight fit blazer. Pointed lapel collar and ...,£169.00,USD,2024-02-19 08:50:07,jackets,MAN
2,180176,End-cap,Yes,Clothing,Yes,2220,Zara,https://www.zara.com/us/en/slim-fit-suit-jacke...,335342680-800-44,SLIM FIT SUIT JACKET,Slim fit jacket. Notched lapel collar. Long sl...,£129.00,USD,2024-02-19 08:50:07,jackets,MAN
3,112917,Aisle,Yes,Clothing,Yes,1568,Zara,https://www.zara.com/us/en/stretch-suit-jacket...,328303236-420-44,STRETCH SUIT JACKET,Slim fit jacket made of viscose blend fabric. ...,£129.00,USD,2024-02-19 08:50:08,jackets,MAN
4,192936,End-cap,No,Clothing,Yes,2942,Zara,https://www.zara.com/us/en/double-faced-jacket...,312368260-800-2,DOUBLE FACED JACKET,Jacket made of faux leather faux shearling wit...,£139.00,USD,2024-02-19 08:50:08,jackets,MAN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
247,159182,Front of Store,Yes,Clothing,No,1014,Zara,https://www.zara.com/us/en/basic-100-wool-swea...,321993245-500-2,FAUX LEATHER OVERSIZED JACKET LIMITED EDITION,Jacket made of technical fabric with padded in...,£169.00,USD,2024-02-19 09:10:44,jackets,MAN
248,199233,Aisle,Yes,Clothing,No,2222,Zara,https://www.zara.com/us/en/colorblock-knit-cro...,330590505-500-2,CONTRASTING PATCHES BOMBER JACKET,Oversized jacket. Notched lapel collar and lon...,£159.00,USD,2024-02-19 09:10:44,jackets,MAN
249,137044,Aisle,No,Clothing,Yes,2534,Zara,https://www.zara.com/us/en/hooded-technical-ja...,320680326-107-39,PATCH BOMBER JACKET,Varsity jacket with elastic collar and long sl...,£12.99,USD,2024-02-19 09:10:45,jackets,MAN
250,154736,Front of Store,Yes,Clothing,Yes,1466,Zara,https://www.zara.com/us/en/houndstooth-suit-ja...,311292244-800-39,CROPPED BOMBER JACKET LIMITED EDITION,Varsity jacket with padded interior. Rib elast...,£19.90,USD,2024-02-19 09:10:46,jackets,MAN


In [3]:
df.head()

,Product_ID,Product_Position,Promotion,Product_Category,Seasonal,Sales_Volume,brand,url,sku,name,description,price,currency,scraped_at,terms,section
0,185102,Aisle,No,Clothing,No,2823,Zara,https://www.zara.com/us/en/basic-puffer-jacket...,272145190-250-2,BASIC PUFFER JACKET,Puffer jacket made of tear-resistant ripstop f...,£19.99,USD,2024-02-19 08:50:06,jackets,MAN
1,188771,Aisle,No,Clothing,No,654,Zara,https://www.zara.com/us/en/tuxedo-jacket-p0889...,324052738-800-46,TUXEDO JACKET,Straight fit blazer. Pointed lapel collar and ...,£169.00,USD,2024-02-19 08:50:07,jackets,MAN
2,180176,End-cap,Yes,Clothing,Yes,2220,Zara,https://www.zara.com/us/en/slim-fit-suit-jacke...,335342680-800-44,SLIM FIT SUIT JACKET,Slim fit jacket. Notched lapel collar. Long sl...,£129.00,USD,2024-02-19 08:50:07,jackets,MAN
3,112917,Aisle,Yes,Clothing,Yes,1568,Zara,https://www.zara.com/us/en/stretch-suit-jacket...,328303236-420-44,STRETCH SUIT JACKET,Slim fit jacket made of viscose blend fabric. ...,£129.00,USD,2024-02-19 08:50:08,jackets,MAN
4,192936,End-cap,No,Clothing,Yes,2942,Zara,https://www.zara.com/us/en/double-faced-jacket...,312368260-800-2,DOUBLE FACED JACKET,Jacket made of faux leather faux shearling wit...,£139.00,USD,2024-02-19 08:50:08,jackets,MAN


In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 252 entries, 0 to 251
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Product_ID        252 non-null    int64 
 1   Product_Position  252 non-null    object
 2   Promotion         252 non-null    object
 3   Product_Category  252 non-null    object
 4   Seasonal          252 non-null    object
 5   Sales_Volume      252 non-null    int64 
 6   brand             252 non-null    object
 7   url               252 non-null    object
 8   sku               252 non-null    object
 9   name              251 non-null    object
 10  description       250 non-null    object
 11  price             252 non-null    object
 12  currency          252 non-null    object
 13  scraped_at        252 non-null    object
 14  terms             252 non-null    object
 15  section           252 non-null    object
dtypes: int64(2), object(14)
memory usage: 31.6+ KB


In [5]:
print(df.describe())

          Product_ID  Sales_Volume
count     252.000000    252.000000
mean   153370.503968   1823.702381
std     26160.444549    697.703748
min    110075.000000    529.000000
25%    131053.750000   1243.000000
50%    151681.500000   1839.500000
75%    175669.750000   2398.750000
max    199631.000000   2989.000000


# Data Cleaning

Before performing exploratory data analysis, the dataset is cleaned to ensure consistency, accuracy, and readability.

The cleaning process includes:

1. Renaming columns for clarity  
2. Dropping irrelevant columns  
3. Fixing column text and data types  
4. Handling missing values  
5. Preparing columns for analysis and visualization

### Step 1: Rename Columns

- Column names are simplified for readability and easier referencing in code.  
- **Example:** `Product_Category`, `Sales_Volume`, `Product_Position` have readable and consistent names.


In [6]:
df.rename(columns={
    'Product_ID': 'Product_ID',
    'Product_Position': 'Product_Position',
    'Product_Category': 'Product_Category',
    'Sales_Volume': 'Sales',
    'Price': 'Price',
    'Promotion': 'Promotion',
    'Seasonal': 'Seasonal',
    'Name': 'Name'
}, inplace=True)

df.columns

Index(['Product_ID', 'Product_Position', 'Promotion', 'Product_Category',
       'Seasonal', 'Sales', 'brand', 'url', 'sku', 'name', 'description',
       'price', 'currency', 'scraped_at', 'terms', 'section'],
      dtype='object')

### Step 2: Drop Irrelevant Columns

- Columns not needed for this analysis are removed to streamline the dataset.  
- **Example:** Dropping `brand`, `url`, `sku`, `description` since they don’t contribute to sales insights.

In [7]:
df = df[['Product_Category', 'Sales', 'price',
         'Promotion', 'Seasonal', 'Product_Position', 'name']]
print(df.head())

  Product_Category  Sales    price Promotion Seasonal Product_Position  \
0         Clothing   2823   £19.99        No       No            Aisle   
1         Clothing    654  £169.00        No       No            Aisle   
2         Clothing   2220  £129.00       Yes      Yes          End-cap   
3         Clothing   1568  £129.00       Yes      Yes            Aisle   
4         Clothing   2942  £139.00        No      Yes          End-cap   

                   name  
0   BASIC PUFFER JACKET  
1         TUXEDO JACKET  
2  SLIM FIT SUIT JACKET  
3   STRETCH SUIT JACKET  
4   DOUBLE FACED JACKET  


In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 252 entries, 0 to 251
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Product_Category  252 non-null    object
 1   Sales             252 non-null    int64 
 2   price             252 non-null    object
 3   Promotion         252 non-null    object
 4   Seasonal          252 non-null    object
 5   Product_Position  252 non-null    object
 6   name              251 non-null    object
dtypes: int64(1), object(6)
memory usage: 13.9+ KB


### Step 3: Fix Column Text and Data Types

- Ensuring numeric and categorical columns have the correct data types and consistent text formatting.  
- **Example:** Converting `Sales_Volume` and `Price` to numeric, and `Promotion` or `Product_Position` to categorical types.

- Verify the dataset is clean and ready for analysis using:
  - `df.head()` for a quick preview  
  - `df.info()` to check rows, columns, and data types  
  - `df.describe()` for summary statistics

In [94]:
df['Sales'] = pd.to_numeric(df['Sales'], errors='coerce')

if df['price'].dtype == 'object':
    df['price'] = pd.to_numeric(df['price'].str.replace('£', '', regex=False).str.replace('$', '', regex=False), errors='coerce')
else:
    df['price'] = pd.to_numeric(df['price'], errors='coerce')

categorical_cols = ['Promotion', 'Seasonal', 'Product_Position', 'Product_Category']
for col in categorical_cols:
    df[col] = df[col].astype('category')

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 252 entries, 0 to 251
Data columns (total 8 columns):
 #   Column            Non-Null Count  Dtype   
---  ------            --------------  -----   
 0   Product_Category  252 non-null    category
 1   Sales             252 non-null    int64   
 2   price             252 non-null    float64 
 3   Promotion         252 non-null    category
 4   Seasonal          252 non-null    category
 5   Product_Position  252 non-null    category
 6   Sub_Category      251 non-null    object  
 7   Price_Range       252 non-null    category
dtypes: category(5), float64(1), int64(1), object(1)
memory usage: 8.0+ KB


In [10]:
df.head()

,Product_Category,Sales,price,Promotion,Seasonal,Product_Position,name
0,Clothing,2823,19.99,No,No,Aisle,BASIC PUFFER JACKET
1,Clothing,654,169.00,No,No,Aisle,TUXEDO JACKET
2,Clothing,2220,129.00,Yes,Yes,End-cap,SLIM FIT SUIT JACKET
3,Clothing,1568,129.00,Yes,Yes,Aisle,STRETCH SUIT JACKET
4,Clothing,2942,139.00,No,Yes,End-cap,DOUBLE FACED JACKET


In [11]:
df.describe()

,Sales,price
count,252.000000,252.000000
mean,1823.702381,86.252540
std,697.703748,52.083205
min,529.000000,7.990000
25%,1243.000000,49.900000
50%,1839.500000,79.900000
75%,2398.750000,109.000000
max,2989.000000,439.000000


### Step 4: Handle Missing Values

- Identify missing or null values in critical columns.  
- **Example:** Removing the rows where `Sales_Volume` or `Price` are missing to avoid skewing the results.

In [12]:
df = df.dropna(subset=['Sales', 'price'])
print(df[['Sales', 'price']].isna().sum())

Sales    0
price    0
dtype: int64


### Step 5: Prepare Columns for Analysis and Visualization

- Standardize text and unify categories to make the dataset ready for plotting.  
- **Example:** Standardizing `Product_Position` entries to avoid duplicates like “aisle” vs “Aisle”.

In [13]:
df['Product_Position'] = df['Product_Position'].str.title().str.strip()
df['Promotion'] = df['Promotion'].str.title().str.strip()

df.head()

,Product_Category,Sales,price,Promotion,Seasonal,Product_Position,name
0,Clothing,2823,19.99,No,No,Aisle,BASIC PUFFER JACKET
1,Clothing,654,169.00,No,No,Aisle,TUXEDO JACKET
2,Clothing,2220,129.00,Yes,Yes,End-Cap,SLIM FIT SUIT JACKET
3,Clothing,1568,129.00,Yes,Yes,Aisle,STRETCH SUIT JACKET
4,Clothing,2942,139.00,No,Yes,End-Cap,DOUBLE FACED JACKET


### Step 6: Create Sub-Category Column

- The original `Product_Category` column is not as informative (everything is “Clothing”), so a generated `Sub_Category` column was made based on product names.  

- **Example Sub-Categories:** Jackets, Shirts, Sweaters, Pants, Dresses/Skirts, Suits/Blazers

In [14]:
def categorize_product(name):
    name = str(name).upper()
    if "JACKET" in name or "BOMBER" in name or "PUFFER" in name:
        return "Jacket"
    elif "SHIRT" in name or "POLO" in name or "T-SHIRT" in name:
        return "Shirt"
    elif "SWEATER" in name or "KNIT" in name or "CARDIGAN" in name:
        return "Sweater"
    elif "PANTS" in name or "JEANS" in name or "TROUSERS" in name:
        return "Pants"
    elif "DRESS" in name or "SKIRT" in name:
        return "Dress/Skirt"
    elif "SUIT" in name or "BLAZER" in name:
        return "Suit/Blazer"
    else:
        return "Other"

df['Sub_Category'] = df['name'].apply(categorize_product)

print(df[['name', 'Sub_Category']].head())

                   name Sub_Category
0   BASIC PUFFER JACKET       Jacket
1         TUXEDO JACKET       Jacket
2  SLIM FIT SUIT JACKET       Jacket
3   STRETCH SUIT JACKET       Jacket
4   DOUBLE FACED JACKET       Jacket


### Step 7: Drop `name` Column

- The `name` column contains specific product names, which are too granular for this analysis. Since there is now have a `Sub_Category` column, `name` is redundant. Dropping it will simplify the dataset and make the analysis easier.

In [15]:
df = df.drop(columns=['name'])

print(df.head())

  Product_Category  Sales   price Promotion Seasonal Product_Position  \
0         Clothing   2823   19.99        No       No            Aisle   
1         Clothing    654  169.00        No       No            Aisle   
2         Clothing   2220  129.00       Yes      Yes          End-Cap   
3         Clothing   1568  129.00       Yes      Yes            Aisle   
4         Clothing   2942  139.00        No      Yes          End-Cap   

  Sub_Category  
0       Jacket  
1       Jacket  
2       Jacket  
3       Jacket  
4       Jacket  


In [16]:
#Final Check of Data Set
df

,Product_Category,Sales,price,Promotion,Seasonal,Product_Position,Sub_Category
0,Clothing,2823,19.99,No,No,Aisle,Jacket
1,Clothing,654,169.00,No,No,Aisle,Jacket
2,Clothing,2220,129.00,Yes,Yes,End-Cap,Jacket
3,Clothing,1568,129.00,Yes,Yes,Aisle,Jacket
4,Clothing,2942,139.00,No,Yes,End-Cap,Jacket
...,...,...,...,...,...,...,...
247,Clothing,1014,169.00,Yes,No,Front Of Store,Jacket
248,Clothing,2222,159.00,Yes,No,Aisle,Jacket
249,Clothing,2534,12.99,No,Yes,Aisle,Jacket
250,Clothing,1466,19.90,Yes,Yes,Front Of Store,Jacket


---

# Exploratory Data Analysis (EDA) — Product Placement for Higher Sales

Question 1: Which product placements lead to higher sales?

Focusing on:

- Total and average sales by product placement and whether products on aisles, end-caps, or front-of-store locations sell better.
- Records per product placement to see how products are distributed across different store locations.

These tables provide a baseline view of how product placement influences sales before moving into deeper analysis and visualizations.

In [17]:
placement_sales_summary = df.groupby('Product_Position')['Sales'].agg(['sum', 'mean']).reset_index()
placement_sales_summary = placement_sales_summary.rename(columns={'sum': 'Total_Sales', 'mean': 'Average_Sales'})

print("Total and Average Sales by Product Placement:")
print(placement_sales_summary)

placement_counts = df['Product_Position'].value_counts().reset_index()
placement_counts.columns = ['Product_Position', 'Record_Count']

print("\nNumber of Records per Product Placement:")
print(placement_counts)

Total and Average Sales by Product Placement:
  Product_Position  Total_Sales  Average_Sales
0            Aisle       177396    1828.824742
1          End-Cap       152930    1778.255814
2   Front Of Store       129247    1873.144928

Number of Records per Product Placement:
  Product_Position  Record_Count
0            Aisle            97
1          End-Cap            86
2   Front Of Store            69


#### Product Placement Analysis

Summary of key trends in Zara's sales, looking at total and average sales by placement, as well as product distribution across store locations.

Key insights include:

1. Aisle products generate the highest total sales (177,396), primarily because there are more products located in aisles (97 records).
2. Front Of Store products have the highest average sales per product (1,873), even though there are fewer items (69 records), indicating that prime placement boosts individual product performance.
3. End-Cap placements perform moderately in both total and average sales (152,930 total; 1,778 average), serving as a balance between volume and visibility.
4. Overall, product placement significantly affects sales. More products on aisles maximize total revenue, while Front Of Store placement maximizes sales per item.

---

# Exploratory Data Analysis (EDA) — Sale Drivers Summary

Question 2: Which products and categories drive the most sales?

Focusing on:

- Total and average sales by sub-category
- Identify which types of products (jackets, shirts, etc.) perform best.
- Records per sub-category, how many products were promoted versus non-promoted.

In [18]:
print("Total and Average Sales by Sub-Category:")
subcategory_sales = (
    df.groupby('Sub_Category')['Sales']
      .agg(Total_Sales='sum', Average_Sales='mean')
      .reset_index()
)
subcategory_sales['Average_Sales'] = subcategory_sales['Average_Sales'].round(2)
print(subcategory_sales.sort_values(by='Total_Sales', ascending=False))

print("\nNumber of Records per Sub-Category:")
subcategory_counts = (
    df['Sub_Category']
      .value_counts()
      .reset_index()
      .rename(columns={'index':'Sub_Category', 'Sub_Category':'Record_Count'})
)
print(subcategory_counts)

print("\nPromotion Distribution by Sub-Category:")
promotion_counts = (
    df.groupby(['Sub_Category', 'Promotion'])
      .size()
      .reset_index(name='Count')
)
print(promotion_counts)

Total and Average Sales by Sub-Category:
  Sub_Category  Total_Sales  Average_Sales
0       Jacket       222848        1841.72
3        Shirt        92102        1771.19
4      Sweater        72505        1812.62
1        Other        58645        1891.77
2        Pants        11635        1662.14

Number of Records per Sub-Category:
  Record_Count  count
0       Jacket    121
1        Shirt     52
2      Sweater     40
3        Other     31
4        Pants      7

Promotion Distribution by Sub-Category:
  Sub_Category Promotion  Count
0       Jacket        No     63
1       Jacket       Yes     58
2        Other        No     15
3        Other       Yes     16
4        Pants        No      5
5        Pants       Yes      2
6        Shirt        No     27
7        Shirt       Yes     25
8      Sweater        No     22
9      Sweater       Yes     18


####Product and Category Sales Analysis

Summary of total and average sales by sub-category, number of products, and promotion distribution.

Key insights include:

1. Jackets are the top-performing sub-category, generating the highest total sales (222,848) and representing the largest number of products (121 records).
2. Shirts and sweaters also contribute significantly to total sales (92,102 and 72,505), with average sales per product ranging from 1,771 to 1,812, showing solid performance despite fewer items than jackets.
3. Other products and pants have fewer records and lower total sales, but some “Other” items achieve the highest average sales per product (1,891), indicating that certain niche items sell well individually. (Could be due to seasons)
4. Sales are fairly balanced between promoted and non-promoted items within each sub-category. Jackets, shirts, and sweaters show nearly equal counts of promoted vs. non-promoted products, suggesting promotions are applied across categories rather than concentrated in one.

---

# Exploratory Data Analysis (EDA) — Pricing Summary

Question 3: How does pricing impact sales?

Focusing on:

- Average sales by price range and understading whether higher or lower-priced products sell better.
- Price distribution across sub-categories help identify which products fall into which price brackets.


In [19]:
bins = [0, 50, 100, 150, 200, float('inf')]
labels = ['0-50', '51-100', '101-150', '151-200', '200+']

#Price Range was added for price bins
df['Price_Range'] = pd.cut(df['price'], bins=bins, labels=labels, right=True)

price_summary = df.groupby('Price_Range', observed=True)['Sales'].agg(
    Total_Sales='sum',
    Average_Sales='mean'
).reset_index()

price_summary['Average_Sales'] = price_summary['Average_Sales'].round(2)

print("Sales Summary by Price Range:")
print(price_summary)

Sales Summary by Price Range:
  Price_Range  Total_Sales  Average_Sales
0        0-50       127861        1800.86
1      51-100       200553        1874.33
2     101-150        92366        1811.10
3     151-200        32735        1722.89
4        200+         6058        1514.50


#### Pricing Analysis

Summary looking into the total and average sales across different price ranges.

Key insights include:

1. Products priced 51-100 generate the highest total sales (200,553) and also the highest average sales per item (1,874), suggesting this is the sweet spot for customers.
2. Lower-priced items (0-50) also sell well overall (127,861 total), with solid average sales (1,801), showing a strong volume for budget-friendly products.
3. Mid-to-high priced items (101-150 and 151-200) see declining total and average sales, showing that fewer customers purchase more expensive items.
4. Products over 200+ have the lowest total (6,058) and average sales (1,515), suggesting very high-priced items have limited demand.
5. Overall, moderately priced products drive the most sales, while very high-priced items sell less frequently.

---

# Exploratory Data Analysis (EDA) — Promotions Summary

Question 4: Do promotions boost sales?

Focusing on:

- Average sales for promoted vs non-promoted products can tell whether or not promotions increase sales.
- Records by promotion status can show how many products were promoted versus non-promoted.

In [20]:
promo_summary = df.groupby('Promotion')['Sales'].agg(
    Total_Sales='sum',
    Average_Sales='mean',
    Record_Count='count'
).reset_index()

promo_summary['Average_Sales'] = promo_summary['Average_Sales'].round(2)

print("Sales Summary by Promotion Status:")
print(promo_summary)

Sales Summary by Promotion Status:
  Promotion  Total_Sales  Average_Sales  Record_Count
0        No       240312        1820.55           132
1       Yes       219261        1827.18           120


####Promotions Analysis

Summary of the total and average sales for promoted vs non-promoted products, as well as the distribution of promoted items.

Key insights include:

1. Non-promoted products generate slightly higher total sales (240,312) than promoted products (219,261), largely because there are more non-promoted items (132 records vs 120).
2. Promoted products have a slightly higher average sales per item (1,827) compared to non-promoted items (1,820), suggesting promotions may boost individual product performance.
3. The difference is relatively small, indicating that while promotions can help, the effect might be limited or depend on other factors like product type or placement.

---

# Exploratory Data Analysis (EDA) — Seasonal Summary

Question 5: How does seasonality affect product performance?

Focusing on:

- Average sales for seasonal vs non-seasonal products and if seasonal products sell differently.
- Records by seasonality to check how many products are marked as seasonal.
- Showing which products based on catergory sell more or less during seasonal times


In [21]:
print("Sales Summary by Seasonality:")

seasonal_summary = (
    df.groupby('Seasonal', observed=True)['Sales']
      .agg(Total_Sales='sum', Average_Sales='mean', Record_Count='count')
      .reset_index()
)
seasonal_summary['Average_Sales'] = seasonal_summary['Average_Sales'].round(2)

print(seasonal_summary)

Sales Summary by Seasonality:
  Seasonal  Total_Sales  Average_Sales  Record_Count
0       No       226392        1825.74           124
1      Yes       233181        1821.73           128


In [22]:
#Seasonal Sales By Sub_Category

In [23]:
seasonal_subcat_summary = (
    df.groupby(['Sub_Category', 'Seasonal'], observed=True)['Sales']
      .agg(Total_Sales='sum', Average_Sales='mean', Record_Count='count')
      .reset_index()
)
seasonal_subcat_summary['Average_Sales'] = seasonal_subcat_summary['Average_Sales'].round(2)

print("Sales Summary by Sub-Category and Seasonality:")
print(seasonal_subcat_summary)

Sales Summary by Sub-Category and Seasonality:
  Sub_Category Seasonal  Total_Sales  Average_Sales  Record_Count
0       Jacket       No       110980        1849.67            60
1       Jacket      Yes       111868        1833.90            61
2        Other       No        39803        1990.15            20
3        Other      Yes        18842        1712.91            11
4        Pants       No         7987        1996.75             4
5        Pants      Yes         3648        1216.00             3
6        Shirt       No        45514        1685.70            27
7        Shirt      Yes        46588        1863.52            25
8      Sweater       No        20270        1689.17            12
9      Sweater      Yes        52235        1865.54            28


#### Seasonal Sales Analysis by Product Type

Summary looking into total and average sales by seasonality, broken down by product type.

Key insights include:

1. Jackets sell almost equally in seasonal and non-seasonal categories (111k total sales each), with slightly higher average sales for non-seasonal jackets (1,849 vs 1,834). This suggests jackets are consistently popular year-round.
2. Shirts and Sweaters perform better when marked as seasonal, showing higher average sales (Shirts: 1,864 vs 1,686; Sweaters: 1,866 vs 1,689), even if total sales are not drastically different.
3. Other products and Pants see lower sales when seasonal. For example, pants’ total sales drop from 7,987 to 3,648 and average sales drop from 1,997 to 1,216.
4. Overall, the impact of seasonality varies by product type: some categories benefit from seasonal promotion (Shirts, Sweaters), while others are less dependent on seasonal trends (Jackets, Pants).

---

#Sales Overview by Product Placement
*Interactive chart showing what in store product placement drive the most sales*

In [91]:
from bokeh.plotting import figure, show
from bokeh.models import ColumnDataSource, HoverTool, NumeralTickFormatter, LinearAxis, Range1d
from bokeh.io import output_notebook
import pandas as pd

output_notebook()

placement_sales_summary = pd.DataFrame({
    'Product_Position': ['Aisle', 'End-Cap', 'Front Of Store'],
    'Total_Sales': [177396, 152930, 129247],
    'Average_Sales': [1828.82, 1778.26, 1873.14],
    'Record_Count': [97, 86, 69]
})

source = ColumnDataSource(placement_sales_summary)

p = figure(
    x_range=placement_sales_summary['Product_Position'],
    height=400,
    width=700,
    title="Sales Overview by Product Placement",
    toolbar_location=None,
    tools="hover",
    tooltips="@Product_Position<br>Total Sales: @Total_Sales{0,0}<br>Average Sales: @Average_Sales{0,0.00}<br>Record Count: @Record_Count",
    y_range=(0, 200000)
)

p.line(
    x='Product_Position',
    y='Total_Sales',
    source=source,
    line_width=3,
    color="#A4968A",
    legend_label="Total Sales",
    line_dash='solid'
)
p.circle(
    x='Product_Position',
    y='Total_Sales',
    source=source,
    size=8,
    color="#A4968A",
    legend_label="Total Sales"
)

p.extra_y_ranges = {"secondary": Range1d(start=0, end=2000)}
p.add_layout(LinearAxis(y_range_name="secondary", axis_label="Average Sales / Record Count"), 'right')

p.line(
    x='Product_Position',
    y='Average_Sales',
    y_range_name="secondary",
    source=source,
    line_width=3,
    color="#8C7B6B",
    legend_label="Average Sales",
    line_dash='dashed'
)
p.circle(
    x='Product_Position',
    y='Average_Sales',
    y_range_name="secondary",
    source=source,
    size=8,
    color="#8C7B6B",
    legend_label="Average Sales"
)

p.line(
    x='Product_Position',
    y='Record_Count',
    y_range_name="secondary",
    source=source,
    line_width=3,
    color="#B3A394",
    legend_label="Record Count",
    line_dash='dotdash'
)
p.circle(
    x='Product_Position',
    y='Record_Count',
    y_range_name="secondary",
    source=source,
    size=8,
    color="#B3A394",
    legend_label="Record Count"
)

p.yaxis[0].formatter = NumeralTickFormatter(format="0,0")
p.yaxis[1].formatter = NumeralTickFormatter(format="0,0")
p.xaxis.axis_label = "Product Placement"
p.grid.grid_line_color = "grey"
p.legend.location = "top_right"
p.legend.click_policy = "hide"

show(p)

#### Insight #1

The Aisle placement generates the highest total sales of 177,396, followed by End-Cap with 152,930, and Front Of Store at 129,247.

#### Insight #2

Average sales per transaction are highest in Front Of Store at 1,873, slightly above Aisle (1,829) and End-Cap (1,778), indicating that while fewer units are sold at the front, the revenue per transaction is higher.

#### Insight #3

The number of records is proportional to sales: Aisle has 97 transactions, End-Cap 86, and Front Of Store 69, suggesting product placement affects transaction volume.

#### Insight #4

This shows that strategic placement in high-traffic areas (Aisle and End-Cap) increases overall revenue, but Front Of Store items achieve higher average sales per transaction, reflecting targeted premium or promotional purchases.

---

#

#Sale Drivers By Category
*Interactive chart showing what category of clothes drive the most sales*

In [86]:
from bokeh.plotting import figure, show
from bokeh.models import ColumnDataSource, HoverTool, NumeralTickFormatter, FixedTicker
from bokeh.io import output_notebook
from bokeh.transform import dodge
import pandas as pd

output_notebook()

subcategory_sales = pd.DataFrame({
    'Sub_Category': ['Jacket', 'Shirt', 'Sweater', 'Other', 'Pants'],
    'Total_Sales': [222848, 92102, 72505, 58645, 11635],
    'Average_Sales': [1841.72, 1771.19, 1812.62, 1891.77, 1662.14],
    'Record_Count': [121, 52, 40, 31, 7]
})

source = ColumnDataSource(subcategory_sales)

metrics = ['Total_Sales', 'Average_Sales', 'Record_Count']

colors = ["#A4968A", "#605046", "#000000"]

p = figure(
    x_range=subcategory_sales['Sub_Category'],
    height=400,
    width=700,
    title="Sales Drivers By Category",
    toolbar_location=None,
    tools="hover",
    tooltips="@Sub_Category<br>Total Sales: @Total_Sales{0,0}<br>Average Sales: @Average_Sales{0,0.00}<br>Record Count: @Record_Count"
)

bar_width = 0.2
offset = [-bar_width, 0, bar_width]

for i, metric in enumerate(metrics):
    p.vbar(
        x=dodge('Sub_Category', offset[i], range=p.x_range),
        top=metric,
        width=bar_width,
        source=source,
        color=colors[i],
        legend_label=metric
    )

p.x_range.range_padding = 0.05
p.xaxis.major_label_orientation = 1
p.xaxis.axis_label = "Sub-Category"
p.yaxis.axis_label = "Value"

p.yaxis.ticker = FixedTicker(ticks=[10000, 50000, 100000, 150000, 200000, 250000])
p.yaxis.formatter = NumeralTickFormatter(format="0,0")

p.grid.grid_line_color = "grey"
p.legend.location = "top_right"
p.legend.click_policy = "hide"

show(p)

#### Insight #1

Jackets dominate total sales with 222,848, followed by Shirts at 92,102 and Sweaters at 72,505. Pants and Other contribute far less, at 11,635 and 58,645.

#### Insight #2

Jackets achieved 222,848 in total sales from 121 transactions, while Pants only generated 11,635 from 7 transactions.

#### Insight #3

Average sales per transaction are highest for Other at 1,891.77 and lowest for Pants at 1,662.14, with Jackets and Sweaters maintaining moderate values around 1,840-1,810.

#### Insight #4

high-selling categories perform well regardless of promotions, whereas smaller categories like Pants and Other may benefit more from promotional activity.

---

# Sales Summary By Price Range
*Interactive chart showing how pricing impacts sales*

In [64]:
from bokeh.plotting import figure, show
from bokeh.models import ColumnDataSource, HoverTool, LinearAxis, Range1d, NumeralTickFormatter
from bokeh.io import output_notebook
output_notebook()

price_summary = pd.DataFrame({
    'Price_Range': ['0-50', '51-100', '101-150', '151-200', '200+'],
    'Total_Sales': [127861, 200553, 92366, 32735, 6058],
    'Average_Sales': [1800.86, 1874.33, 1811.10, 1722.89, 1514.50]
})

source = ColumnDataSource(price_summary)

p = figure(
    x_range=price_summary['Price_Range'],
    height=400,
    width=700,
    title="Sales Summary by Price Range",
    toolbar_location=None
)

p.extra_y_ranges = {
    "total_sales": Range1d(start=0, end=price_summary['Total_Sales'].max() * 1.2)
}
p.add_layout(
    LinearAxis(
        y_range_name="total_sales",
        axis_label="Total Sales",
        formatter=NumeralTickFormatter(format="0,0")
    ),
    'right'
)
p.step(
    x='Price_Range',
    y='Total_Sales',
    source=source,
    y_range_name="total_sales",
    mode='center',
    line_width=3,
    color="#605046",
    legend_label="Total Sales"
)

p.y_range = Range1d(start=0, end=2000)
p.step(
    x='Price_Range',
    y='Average_Sales',
    source=source,
    mode='center',
    line_dash="dashed",
    line_width=3,
    color="#A4968A",
    legend_label="Average Sales"
)

hover = HoverTool(tooltips=[
    ("Price Range", "@Price_Range"),
    ("Total Sales", "@Total_Sales{0,0}"),
    ("Average Sales", "@Average_Sales{0,0.00}")
])
p.add_tools(hover)

p.xaxis.axis_label = "Price Range"
p.yaxis.axis_label = "Average Sales"
p.legend.location = "top_right"
p.legend.click_policy = "hide"
p.grid.grid_line_color = None

show(p)

#### Insight #1

Most sales are concentrated in the mid-price ranges, with the 51-100 range generating the highest total sales of 200,553.

#### Insight #2

The lowest (0-50) and highest (200+) price ranges contribute significantly less in total sales, at 127,861 and 6,058, respectively, showing that extreme price points sell fewer units.

#### Insight #3

Average Sales per transaction peaks in the 51-100 range at 1,874, and gradually decreases as price increases, dropping to 1,514 in the 200+ range.

#### Insight #4

This indicates that while higher-priced items sell far fewer units, the average revenue per transaction decreases only moderately, suggesting consistent purchasing behavior across most price ranges, except the highest range where both sales and average revenue drop.

#Total Sales By Promotion Status
*Interactive chart showing how much sales are made during promotion and sales vs regualar pricing*

In [55]:
from bokeh.plotting import figure, show
from bokeh.models import ColumnDataSource, HoverTool
from bokeh.transform import cumsum
from bokeh.io import output_notebook
import pandas as pd
import math

output_notebook()

promo_summary['Total_Percent'] = promo_summary['Total_Sales'] / promo_summary['Total_Sales'].sum() * 100
promo_summary['angle'] = promo_summary['Total_Sales'] / promo_summary['Total_Sales'].sum() * 2 * math.pi
promo_summary['color'] = ["#605046", "#A4968A"]

promo_summary['label'] = promo_summary.apply(
    lambda row: f"{row['Promotion']}\nTotal: {row['Total_Sales']:,}\nAvg: {row['Average_Sales']:.2f}\n{row['Total_Percent']:.1f}%", axis=1
)

source = ColumnDataSource(promo_summary)

p = figure(
    height=400, width=400,
    title="Total Sales by Promotion Status",
    toolbar_location=None,
    tools="hover, tap",
    tooltips="@label"
)

p.annular_wedge(
    x=0, y=1,
    inner_radius=0.3,
    outer_radius=0.8,
    start_angle=cumsum('angle', include_zero=True),
    end_angle=cumsum('angle'),
    line_color="white",
    fill_color='color',
    legend_field='Promotion',
    source=source,
    selection_color="firebrick",
    nonselection_fill_alpha=0.4
)

p.axis.visible = False
p.grid.grid_line_color = None
p.legend.location = "top_right"
p.legend.label_text_font_size = "10pt"
p.legend.click_policy = "hide"

show(p)

#### Insight #1

Total sales are higher for non-promoted items (240,312) than for promoted items (219,261), suggesting that promotions do not always drive higher overall sales.

#### Insight #2

Despite lower total sales, promoted items have a slightly higher average sales per transaction (1,827.18 vs 1,820.55), indicating that promotions may increase the value of individual transactions.

#### Insight #3

The number of transactions is slightly higher for non-promoted items (132 vs 120), reinforcing that non-promoted products dominate in volume, while promotions improve per-transaction performance.

#### Insight #4

This shows that promotions can enhance the average purchase value but do not necessarily increase total revenue, highlighting the need to balance promotion strategy with volume.

---

# Total Sales By Seasonality  
*Interactive pie chart showing the proportion of total sales in and out of season*

In [47]:
from bokeh.plotting import figure, show
from bokeh.models import ColumnDataSource, HoverTool
from bokeh.transform import cumsum
from bokeh.io import output_notebook
import pandas as pd
import math

output_notebook()

seasonal_summary['Seasonal_Label'] = seasonal_summary['Seasonal'].map({
    'Yes': 'In Season',
    'No': 'Out of Season'
})

seasonal_summary['angle'] = seasonal_summary['Total_Sales'] / seasonal_summary['Total_Sales'].sum() * 2 * math.pi
seasonal_summary['color'] = ["#605046", "#A4968A", "#C7B299", "#E0D8C3"][:len(seasonal_summary)]

source = ColumnDataSource(seasonal_summary)

p = figure(
    height=400, width=400,
    title="Total Sales by Seasonality",
    toolbar_location=None,
    tools="hover",
    tooltips="@Seasonal_Label: @Total_Sales{0,0}"
)

p.wedge(
    x=0, y=1, radius=0.8,
    start_angle=cumsum('angle', include_zero=True),
    end_angle=cumsum('angle'),
    line_color="white",
    fill_color='color',
    legend_field='Seasonal_Label',
    source=source
)

p.axis.visible = False
p.grid.grid_line_color = None
p.legend.location = "top_right"
p.legend.label_text_font_size = "10pt"

show(p)

# Total Sales By Sub Category In & Out Of Season
*Interactive pie chart showing the proportion of total sales by sub-category in or out of season*

In [46]:
from bokeh.plotting import figure, show
from bokeh.models import ColumnDataSource, HoverTool
from bokeh.transform import cumsum
from bokeh.io import output_notebook
import pandas as pd
import math

output_notebook()

seasonal_subcat_summary['Seasonal_Label'] = seasonal_subcat_summary['Seasonal'].map({
    'Yes': 'In Season',
    'No': 'Out of Season'
})

season_labels = seasonal_subcat_summary['Seasonal_Label'].unique()
colors = ["#605046", "#A4968A", "#C7B299", "#E0D8C3", "#B3A394", "#8C7B6B"]

for season in season_labels:
    data = seasonal_subcat_summary[seasonal_subcat_summary['Seasonal_Label'] == season].copy()
    data['angle'] = data['Total_Sales'] / data['Total_Sales'].sum() * 2 * math.pi
    data['color'] = colors[:len(data)]
    source = ColumnDataSource(data)

    p = figure(
        height=400, width=400,
        title=f"Total Sales by Sub-Category ({season})",
        toolbar_location=None,
        tools="hover",
        tooltips="@Sub_Category: @Total_Sales{0,0}"
    )

    p.wedge(
        x=0, y=1, radius=0.8,
        start_angle=cumsum('angle', include_zero=True),
        end_angle=cumsum('angle'),
        line_color="white", fill_color='color',
        legend_field='Sub_Category',
        source=source
    )

    p.axis.visible = False
    p.grid.grid_line_color = None
    p.legend.location = "top_right"
    p.legend.label_text_font_size = "10pt"

    show(p)

#### Insight #1

Total sales are slightly higher “In Season” (233,181) compared to out-of-season (226,392), showing that seasonal demand drives a modest increase in overall revenue.

#### Insight #2

The Jacket sub-category has almost equal sales in-season (111,868) and out-of-season (110,980), indicating steady demand regardless of season, unlike other categories.

#### Insight #3

Other categories, such as Sweater, show strong seasonal variation: in-season sales (52,235) are more than double out-of-season sales (20,270), showing sweaters are highly sensitive to seasonality.

#### Insight #4

Average sales per transaction remain fairly consistent across seasons, around 1,820-1,830, suggesting that while total sales fluctuate with season, the typical transaction value is stable.

#### Insight #5

These pie charts emphasize which sub-categories are more seasonal versus stable, informing inventory management and marketing strategies for seasonal promotions.

---

# Sales Summary by Price Range
*Interactive bar chart showing total sales and average sales by price range, with separate vertical axes for clear comparison.*

In [36]:
from bokeh.plotting import figure, show
from bokeh.models import ColumnDataSource, LinearAxis, Range1d, HoverTool, NumeralTickFormatter
from bokeh.transform import dodge
from bokeh.io import output_notebook

output_notebook()

source = ColumnDataSource(price_summary)

p = figure(
    x_range=price_summary['Price_Range'].astype(str),
    height=400,
    width=700,
    title="Sales Summary by Price Range",
    toolbar_location=None
)

p.vbar(
    x=dodge('Price_Range', -0.2, range=p.x_range),
    top='Total_Sales',
    width=0.4,
    source=source,
    color="#605046",
    legend_label="Total Sales"
)

p.y_range.start = 0
p.y_range.end = price_summary['Total_Sales'].max() * 1.1
p.yaxis.axis_label = "Total Sales"
p.yaxis.formatter = NumeralTickFormatter(format="0,0")

p.extra_y_ranges = {
    "avg_sales": Range1d(start=0, end=price_summary['Average_Sales'].max() * 1.2)
}
p.add_layout(
    LinearAxis(
        y_range_name="avg_sales",
        axis_label="Average Sales",
        formatter=NumeralTickFormatter(format="0,0")
    ),
    'right'
)

p.vbar(
    x=dodge('Price_Range', 0.2, range=p.x_range),
    top='Average_Sales',
    width=0.4,
    source=source,
    color="#A4968A",
    y_range_name="avg_sales",
    legend_label="Average Sales"
)

hover = HoverTool(tooltips=[
    ("Price Range", "@Price_Range"),
    ("Total Sales", "@Total_Sales{0,0}"),
    ("Average Sales", "@Average_Sales{0,0.00}")
])
p.add_tools(hover)

p.xaxis.axis_label = "Price Range"
p.xgrid.grid_line_color = None
p.legend.location = "top_right"
p.legend.orientation = "horizontal"

show(p)

#### Insight #1

Most sales are concentrated in the mid-price ranges, with the 51-100 price range generating the highest total sales of 200,553.

#### Insight #2

• Example: The lowest price range (0-50) generated 127,861 in total sales, while the highest price range (200+) contributed only 6,058.

#### Insight #3

Average sales per transaction remain relatively stable across price ranges, from 1,800-1,874 in the lower ranges, gradually decreasing to 1,514 in the highest range.

#### Insight #4

This shows that although higher-priced items sell fewer units, the average revenue per transaction only slightly decreases, showing moderate consistency in purchase behavior across price ranges.

---

## Key Takeaways

### Product Placement and Sales
- **Aisle placements** lead to the highest total sales and transaction counts, showing that high-traffic areas drive volume.  
- **Front Of Store** placements generate fewer transactions but achieve the **highest average sales per transaction**, indicating strategic placement boosts per-sale revenue.  
- **End-Cap** placements perform moderately on both total and average sales.  

### Product Categories Driving Sales
- **Jackets** are the top-performing sub-category in total sales and number of transactions.  
- **Shirts and Sweaters** show moderate total sales with stable average sales.  
- **Pants** and **Other** have lower total sales, highlighting niche demand segments.  

### Pricing Impact
- Sales are concentrated in the mid-price ranges (51-100), which generate the highest total revenue.  
- Higher-priced items (200+) sell fewer units, but average sales per transaction remain relatively stable, showing consistent spending per customer.  

### Promotion Effectiveness
- Non-promoted items generate higher total sales, but promoted items maintain slightly higher average sales per transaction.  
- Promotions can increase transaction volume without drastically affecting revenue per sale, suggesting they are effective for boosting exposure.  

### Seasonality Effects
- “In season” periods slightly outperform “out of season” in total sales for most sub-categories.  
- **Average sales per transaction remain stable** across seasons, indicating that customer spending per transaction is consistent regardless of timing.  

### Strategic Implications
- Focus on **mid-range pricing** for volume while keeping premium options for higher per-transaction revenue.  
- Allocate **high-traffic placements** to top-selling categories like Jackets for maximum revenue impact.  
- Use promotions strategically to boost visibility and transaction counts.  
- Plan inventory and marketing campaigns around seasonality trends to maintain steady sales throughout the year.

# Thank You